test/e2e/extensions/high_availability_test.go

```diff
@@ -0,0 +1,110 @@
+// Copyright 2023 Google LLC All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package extensions
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"agones.dev/agones/pkg/util/runtime"
+	e2eframework "agones.dev/agones/test/e2e/framework"
+	"github.com/stretchr/testify/assert"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/util/wait"
+)
+
+// Test creating a gameserver when one of the extensions pods is down/deleted
+func TestGameServerHealthyAfterDeletingPodWhileOneExtensionsDown(t *testing.T) {
+	logger := e2eframework.TestLogger(t)
+	ctx := context.Background()
+
+	if !runtime.FeatureEnabled(runtime.FeatureSplitControllerAndExtensions) {
+		logger.Infof("Skip test. SplitControllerAndExtensions feature is not enabled: flag is %v", runtime.FeatureEnabled(runtime.FeatureSplitControllerAndExtensions))
+		return
+	}
+
+	assert.NoError(t, waitForAgonesExtensionsRunning(ctx))
+
+	list, err := getAgoneseExtensionsPods(ctx)
+	logger.Infof("Length of pod list is %v", len(list.Items))
+	if err != nil {
+		t.Fatalf("Could not get list of Extension pods: %v", err)
+	}
+	if len(list.Items) <= 1 {
```

Comment:
Use `assert.Greater`. The message is nice, though, so you can keep:

assert.Greater(list.Items, 1, "Cluster has zero or one extension pods")

Reply:
Oh thats neat, I did not know this was an option.